paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt

```diff
@@ -352,29 +353,19 @@ class Paparazzi @JvmOverloads constructor(
   ) {
     val frameNanos = TIME_OFFSET_NANOS + timeNanos
 
+    // System_Delegate.nanoTime() is nanoTime - previousNanoTime + sChoreographerTime.
+    // Because of that setNanosTime needs to be called with 0 first so that the previous time will be 0 for the actual
+    // call. sChoreographerTime does not matter because doFrame which modifies it is not called by Paparazzi.
+    System_Delegate.setNanosTime(0L)
     // Execute the block at the requested time.
     System_Delegate.setNanosTime(frameNanos)
 
-    val choreographer = Choreographer.getInstance()
-    val areCallbacksRunningField = choreographer::class.java.getDeclaredField("mCallbacksRunning")
-    areCallbacksRunningField.isAccessible = true
+    // should happen before doCallbacks so that compose re-composition happens
+    executeHandlerCallbacks()
+    // 1 is the only callbackType allowed by doCallbacks
+    Choreographer_Delegate.doCallbacks(Choreographer.getInstance(), 1, frameNanos)
```

Comment:
`Choreographer_Delegate` also has a `doFrame` method that updates the choreographer time and runs the callbacks. But I decided against using it for now because `sChoreographerTime` is only updated after the callbacks run which would require changing some tests that rely on the time to be already updated and I wanted to avoid updating the tests for now. It also a bit more complicated to use because `sChoreographerTime` can apparently not become smaller without breaking things (`PaparazziTest.animationCallbacksForStaticSnapshots` was suceeding when run on its own but failed when running all of `PaparazziTest` when I tried using doFrame and starting with `0` in each test).